### PR TITLE
Update ecosystem section

### PR DESCRIPTION
**Pull request by mosabua**
* Separate clients into drivers and applications
* Separate separate into official trinodb drivers and applications
* Update all related rendering and content

Follow up PRs will adjust data source rendering and add-on rendering as well

Replaces https://github.com/trinodb/trino.io/pull/712

Preview is available at https://deploy-preview-739--trino.netlify.app/ecosystem/

### CODE REVIEW

**Comment by mosabua** [comment]
I showed this to @martint , @dain , and @electrum and I am getting this live now so we can showcase in Trino Summit. If anything needs tweaking just let me know .. there are numerous follow ups on the schedule that we want to get in as additional tools and such and the current PR is in good shape.

File: ecosystem/client-drivers-intro.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Client drivers are language specific wrappers around the [client
```
Comment (colebow) [inline ecosystem/client-drivers-intro.md:1]: language-specific

File: ecosystem/client-drivers-intro.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+protocol]({{site.baseurl}}/docs/current/client/client-protocol.html). They
+enable [client application]({{site.baseurl}}/ecosystem/client-application.html)
+developers to connect to Trino, submit SQL queries, and receive the results in
+the programming or scripting language of choice. Client drivers remove the need
+for developers to understand and implement the interactions of the client
+protocol, and enables collaboration of all developers in a specific ecosystem.
```
Comment (colebow) [inline ecosystem/client-drivers-intro.md:7]: `protocol and enable collaboration`
no comma, delete the s

File: ecosystem/client-applications-intro.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+in catalogs]({{site.baseurl}}/ecosystem/index.html#data-sources).
+
+Client applications include command line tools, desktop applications, web-based
+applications, and software-as-a-service solutions. Common features are
+interactive SQL query authoring with editors, rich user interfaces for graphical
+query creation, query running and result rendering, visualizations with charts
+and graphs, reporting, and dashboard creation.
+
+Client applications can process the returned data from Trino for use in data
+pipelines across catalogs and from other data sources to Trino catalogs.
```
Comment (colebow) [inline ecosystem/client-applications-intro.md:12]: this is a bit of a word jumble

**Review by colebow — COMMENTED** [review]
This is an awesome restructuring of this, a few nits, but love this

**Pull request merged**